### PR TITLE
[MIG] contract_sale: Migration to 17.0

### PR DESCRIPTION
**Pull request by jaydeep32**

### CODE REVIEW

**Comment by CRogos** [comment]
Have a look at this commit: 
https://github.com/OCA/social/pull/1298/commits

image

You could add a similar commit to this PR with test-requirements.txt adding:
`odoo-addon-contract @ git+https://github.com/OCA/contract.git@refs/pull/1067/head#subdirectory=setup/contract`

Than the module should be installed by runboat and the tests should also be executed. 
After the review is finished, the commit needs to be deleted before the merge.

Could you also cleanup the commit history according to the migration guide (oca-git-bot, oca-ci, webtranslate)? 
https://github.com/OCA/maintainer-tools/wiki/Merge-commits-in-pull-requests

**Comment by CRogos** [comment]
`ERROR: odoo-addon-contract@ git+https://github.com/OCA/contract.git@refs/pull/1067/head#subdirectory=setup/contract from git+https://github.com/OCA/contract.git@refs/pull/1067/head#subdirectory=setup/contract (from -r test-requirements.txt (line 1)) does not appear to be a Python project: neither 'setup.py' nor 'pyproject.toml' found.`

I think because of the change to .toml files link changed to this... sorry.

`odoo-addon-contract@ git+https://github.com/OCA/contract.git@refs/pull/1067/head`

**Comment by jaydeep32** [comment]
@CRogos I am again I think, getting same error
```
ERROR: odoo-addon-contract@ git+https://github.com/OCA/contract.git@refs/pull/1067/head from git+https://github.com/OCA/contract.git@refs/pull/1067/head (from -r test-requirements.txt (line 1)) does not appear to be a Python project: neither 'setup.py' nor 'pyproject.toml' found.
```

**Comment by CRogos** [comment]
I am so sorry. I just found a working sample here: https://github.com/OCA/sale-workflow/pull/2886/commits/4950f5f5266ab1304d111c29c12ffd83dc8c7ec6

We need to readd "#subdirectory=product_set"

`odoo-addon-contract@ git+https://github.com/OCA/contract.git@refs/pull/1067/head#subdirectory=product_set`

**Comment by rousseldenis** [comment]
@jaydeep32 @CRogos The method to do it is described here: https://github.com/OCA/maintainer-tools/wiki/Use-temporary-reference%28s%29-to-another-pull-request%28s%29

**Review by CRogos — APPROVED** [review]
Code LGTM

You could squash some of the commits: 
https://github.com/OCA/maintainer-tools/wiki/Merge-commits-in-pull-requests#mergesquash-the-commits-generated-by-bots-or-weblate

image

**Comment by jaydeep32** [comment]
> Code LGTM
> 
> You could squash some of the commits: https://github.com/OCA/maintainer-tools/wiki/Merge-commits-in-pull-requests#mergesquash-the-commits-generated-by-bots-or-weblate
> 
> image

Could you please tell me how can I push in this PR once I squash these commits?

**Comment by CRogos** [comment]
you can use "git push -f" to overwrite the remote commit history in **archeti-org:17.0-mig-contract_sale** with your new local one.

**Comment by MohamedOsman7** [comment]
> > Code LGTM
> > You could squash some of the commits: https://github.com/OCA/maintainer-tools/wiki/Merge-commits-in-pull-requests#mergesquash-the-commits-generated-by-bots-or-weblate
> > image
> 
> Could you please tell me how can I push in this PR once I squash these commits?

With the following command: git push --force

**Review by MohamedOsman7 — APPROVED** [review]
Functional & code review LGTM

File: test-requirements.txt
Comment (CRogos) [inline test-requirements.txt:1]: this needs to be removed from the [MIG] commit

**Review by CRogos — CHANGES_REQUESTED** [review]

**Comment by jaydeep32** [comment]
Hello @CRogos , @MohamedOsman7 
would you please recheck this PR? I have been trying to rebase with 17.0 since a few days ago. There was a contract module PR merge, and that module is used as depends on this contract_sale, so I thought I should rebase it. But I might have something wrong here, I see now a lot of commits and not sure if this PR is ok or not.
Sorry for the extra trouble but please let me know if there is an issue I will make a new PR.

**Comment by CRogos** [comment]
@jaydeep32 I am not sure what you did wrong but indeed this doesn't look correct. 
I would suggest you start again and cherry-pick these two commits.
image

If you push to an existing branch after rebase, you need to use -f (force) to rewrite the hole history. Otherwise there might be a merge of both versions created.

**Pull request closed without merging**
